scylla/src/transport/topology.rs

```diff
@@ -469,6 +470,23 @@ impl MetadataReader {
             .retry_fetch_metadata_on_nodes(initial, filtered_known_peers, prev_err)
             .await;
 
+        if let Err(prev_err) = result {
+            // If no known peer is reachable, try falling back to initial contact points, in hope that
+            // there are some hostnames there which will resolve to reachable new addresses.
+            warn!("Failed to establish control connection and fetch metadata on all known peers. Falling back to initial contact points.");
+            let (initial_peers, _hostnames) =
+                resolve_contact_points(&self.initial_known_nodes).await;
+            result = self
+                .retry_fetch_metadata_on_nodes(
+                    initial,
+                    initial_peers
+                        .into_iter()
+                        .map(UntranslatedEndpoint::ContactPoint),
+                    prev_err,
+                )
+                .await;
+        }
+
```

Comment:
Does it make sense to do this in case when this is an initial metadata fetch? E.g. if none of the addresses are reachable? Won't it make the connection process (which will ultimately result in a failure) twice as long?

Reply:
Good catch. Let's `if` the initial fetch case out.